src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]),
```

Comment:
As the PR is currently yes (good catch!). If the `React.Children.only` proposal is adopted – and I think it should be – then it should be `element`, not `node`, so that a `ref` can be injected.